tools/azsdk-cli/docs/specs/6-package-updater.spec.md

```diff
@@ -0,0 +1,697 @@
+# Spec: 6 Package - package-update
+
+## Table of Contents
+
+- [Definitions](#definitions)
+- [Background / Problem Statement](#background--problem-statement)
+- [Goals and Exceptions/Limitations](#goals-and-exceptionslimitations)
+- [Design Proposal](#design-proposal)
+- [Open Questions](#open-questions)
+- [Success Criteria](#success-criteria)
+- [Agent Prompts](#agent-prompts)
+- [CLI Commands](#cli-commands)
+- [Implementation Plan](#implementation-plan)
+- [Testing Strategy](#testing-strategy)
+- [Documentation Updates](#documentation-updates)
+- [Metrics/Telemetry](#metricstelemetry)
+
+---
+
+## Definitions
+
+- **Stage 6**: The standardized SDK preparation stage after code/tests/samples validation and before creating the release preparation PR.
+- **Metadata and Documentation Update**: Consolidated set of actions to ensure non-source assets (README, changelog, CI config, version numbers, additional descriptors) are accurate and consistent.
+- **Tool**: A discrete CLI (and MCP agent-exposed) command implementing one cohesive post-generation preparation task.
+- **Inline Script Command**: A single shell command provided directly in config that overrides a script path.
+
+---
+
+## Background / Problem Statement
+
+### Current State
+
+After SDK code generation, build, tests, and sample validation, engineers must perform repetitive cross-language housekeeping: updating changelogs, synchronizing README and language-specific files, adjusting package versions, exporting API surfaces (.NET), and inserting or validating CI configuration. Today these tasks are:
+
+- Partially automated in ad-hoc scripts per language / repo.
+- Inconsistent between management-plane and data-plane SDKs (especially version bump timing and changelog handling).
+- Error-prone when performed manually (missed version increments, stale README examples, inconsistent metadata presented across different files, like `pom.xml`, `_metadata.json`).
+- A barrier for integrating an AI agent workflow because required parameters and step order are not encapsulated as composable tools.
+
+### Why This Matters
+
+Without a unified tool layer:
+
+- Contributors spend time re-learning differing per-language conventions.
+- CI or release pipelines fail late due to missing metadata or mismatched versions.
+- Agents cannot reliably assist because state transitions ("now bump version", "now update changelog") aren't formalized.
+- Management-plane vs data-plane discrepancies introduce confusion (e.g., when to bump versions or generate changelog entries).
```

Comment:
Could you share more about the difference on how data plane SDK do version bump and generate changelog?  is this done manually or using a another tool?